src/core/ledger/distributionSummary/allocationSummary.js

```diff
@@ -0,0 +1,68 @@
+// @flow
+
+import {Ledger} from "../ledger";
+import {type Distribution} from "../distribution";
+import {type Allocation} from "../grainAllocation";
+import {type GrainReceipt} from "../grainAllocation";
+import * as G from "../grain";
+import sortBy from "../../../util/sortBy";
+import {type IdentityId} from "../../identity";
+import * as NullUtil from "../../../util/null";
+import {type CurrencyDetails} from "../../../api/currencyConfig";
+import {formatCenter} from "./distributionSummary";
+
+export function allocationMarkdownSummary(
```

Comment:
this is called `allocationMarkdownSummary` but really would be better called `logAllocationMarkdownSummary`, since it's logging to console, not just producing a summary.

i suggest separating generating the summary table string from actually logging it to console. i.e. make the `allocationMarkdownSummary` just return a string and then the cli code can call `console.log()`.

the big advantage is in testability: you can include some snapshot tables via `expect(allocationMarkdownSummary(x)).toMatchSnapshot()` or `.toMatchInlineSnapshot()`.

Reply:
Done, now returns string.